apps/heft/src/cli/HeftCommandLineUtilities.ts

```diff
@@ -87,12 +71,15 @@ export interface IRegisterParameterOptions extends IBaseCommandLineDefinition {
  * The options object provided to the command line parser when registering a parameter
  * in addition to the action names used to associate the parameter with.
  */
-export interface IRegisterParameterWithArgumentOptions extends IBaseCommandLineDefinitionWithArgument {
-  /**
-   * A string list of one or more action names to associate the parameter with.
-   */
-  associatedActionNames: string[];
-}
+export type IRegisterParameterOptions = IBaseCommandLineDefinition & IParameterAssociatedActionNames;
```

Comment:
ts-command-line's `IBaseCommandLineDefinition` is not a complete definition.  For example, to define a choice parameter, its options look like this:

```ts
export interface ICommandLineChoiceDefinition extends IBaseCommandLineDefinition {
  /**
   * A list of strings (which contain no spaces), of possible options which can be selected
   */
  alternatives: string[];

  /**
   * {@inheritDoc ICommandLineStringDefinition.defaultValue}
   */
  defaultValue?: string;

  /**
   * An optional callback that provides a list of custom choices for tab completion.
   * @remarks
   * This option is only used when `ICommandLineParserOptions.enableTabCompletionAction`
   * is enabled.
   */
  completions?: () => Promise<string[]>;
}
```

(This PR skirts around that problem by only exposing "simple" parameter types, but I feel like we might be painting ourselves into a corner if people ask for more parameter types in the future, or if `ts-command-line` adds more  options for the child interfaces.)

If our approach is to reuse the `ts-command-line` API but add an `associatedActionNames` member, then maybe we could just as easily express that like this:

```ts
// option A
registerFlagParameter(options: ICommandLineFlagDefinition & IHeftRegisterParameterOptions): IHeftFlagParameter;
registerIntegerParameter(options: ICommandLineIntegerDefinition & IHeftRegisterParameterOptions): IHeftIntegerParameter;
```

...or perhaps:

```ts
// option B
registerFlagParameter(options: IHeftRegisterParameterOptions<ICommandLineFlagDefinition >): IHeftFlagParameter;
registerIntegerParameter(options: IHeftRegisterParameterOptions<ICommandLineIntegerDefinition >): IHeftIntegerParameter;

// ...with:

interface ICommandLineIntegerDefinition<T extends IBaseCommandLineDefinition> {
  associatedActionNames: string[];
  definition: T;
}
```

or if we want to keep it super simple:
 
```ts
// option C
registerFlagParameter(associatedActionNames: string[], definition: ICommandLineFlagDefinition): IHeftFlagParameter;
registerIntegerParameter(associatedActionNames: string[], definition: ICommandLineIntegerDefinition): IHeftIntegerParameter;
```

Reply:
I chose option D? Let me know if this works for now.